Updating recommended email settings for monolog

jorgelbg opened the pull request:
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | >=2.3 |
| Fixed tickets | #6467 |

jorgelbg committed:
Updating recommended email settings for monolog

javiereguiluz commented on cookbook/logging/monolog_email.rst:
```diff
@@ -92,26 +97,29 @@ it is broken down.
-                'buffered' => array(
-                    'type'    => 'buffer',
-                    'handler' => 'swift',
+                'duplicated' => array(
```
Should be `deduplicated` here?

xabbuh commented on cookbook/logging/monolog_email.rst:
```diff
@@ -49,33 +51,36 @@ it is broken down.
                 action-level="error"
                 And add this child inside this monolog:handler
-                <monolog:excluded-404>^/</monolog:excluded-404>
+                <monolog:excluded-404>^/</monolog:excluded-404> 
```
Can you revert this change (the extra space)?

Txumari commented:
I'm new here, but I think should be change the buffered explanation with the new handler deduplicated (The ``deduplicated`` handler simply keeps...)

jorgelbg commented:
Fixed the typos, and amended the commit. Thanks @javiereguiluz @xabbuh

xabbuh commented:
👍

javiereguiluz commented:
👍

(by the way, I really dislike the "deduplication" word ... but that's something we cannot change)

WouterJ commented:
I agree with @Txumari here, after this code block we talk about the buffered handler a couple times. We should change it to deduplicated handler (and also change the description).

jorgelbg commented:
Should we mantain some information about the `buffer` handler? its still a valid handler according to https://github.com/Seldaek/monolog/blob/master/doc/02-handlers-formatters-processors.md#wrappers--special-handlers but honestly the `deduplication` handler does also the buffering + deduplication of messages. The final example use the `group` and `buffer` ?

jorgelbg committed:
Changing explanation of the deduplicated/deduplication handlers and removing some mentions of the buffer handler
Fixing some syntax issues

jorgelbg commented:
I've updated the explanation and fixed the mentions of the `buffered` handler, also changed the last example in the doc, that talked about combining the handlers to store the messages on the server and also send the logs by email.

xabbuh commented on cookbook/logging/monolog_email.rst:
```diff
@@ -121,20 +129,23 @@ it is only triggered when the action level, in this case ``critical`` is reached
+unique over a given period of time (60 seconds by default). If the records are
+duplicates they are simply discarded. Adding this handler reduces the amount of
+notifications to a manageable level, specially in critical failure scenarios.
+The messages are then passed to the ``swift`` handler. This is the handler that
```
I would add a blank line before this sentence to have it rendered as its own paragraph to improve readability.

jorgelbg replied:
Done!

jorgelbg committed:
Adding a new line to improve readability

WouterJ commented:
:+1: I like it

Status: Reviewed

xabbuh commented:
Thank you @jorgelbg.
